### ISSUE TITLE
Does MMT4 randomise the data before training ?

### ISSUE THREAD

**Issue by EtienneAb3d** [modernmt/modernmt#478]
I'm currently training MMT4 on a much larger dataset than what I usually provide to MMT3. Each epoch seems to take several days (and even perhaps several weeks ?). The training curves are already very flat. But, when testing the intermediate model, it seems to be of very lower quality than what I usually get with MMT3 in the same time (with smaller datasets).
Of course, for the moment, I don't know why this seems to be of lower quality. 
But, I think of a problem: if the dataset is not randomised, with large corpus provided, the training data used in the first time may be of a special kind, very different of the data that will come later in the training. It would be nice to have them randomised to perform an uniform training.

**Comment by davidecaroselli** [comment]
Yes, data is randomized (it is actually done by Fairseq).

Even if fairseq still reports "epochs", you can ignore that and focus on number of updates. We have huge training set (billions of words) training for "only" 2 epochs and they are outperforming other online generic engines.

In terms of quality, that's strange but let's wait for the training end. Are the models equals (transformer big and transformer big)? If you have any doubts, try comparing apples with apples, in other words the same training set without even the cleaning. We gained more than 2 BLEU points in average over previous version of ModernMT thanks to new cleaning pipeline and Fairseq.

**Comment by davidecaroselli** [comment]
Hi @EtienneAb3d 

we have finally found out the reason for the performance degradation: the default "base" model was not exactly the same as the V3 "base" model, but smaller. Moreover there were some training parameters (such as learning rate) that we have now optimized.

The final results show an improvement over v3.3 on the IWSLT task, and I confirm that on much more larger tasks, we measure a huge improvement over the previous version.

Here is the issue with the results for v4 vs v3.3: https://github.com/modernmt/modernmt/issues/485

Thanks for the support and patience!
